### PR TITLE
DOCS 2.4 : Fixed all internally generated 404s

### PR DESCRIPTION
**Pull request by SpiritLevel**
DOCS 2.4 : Fixed some but not all external links causing 404s

DOCS 2.4 : adjusted link labels

DOCS 2.4 : another link label fixed up

### CODE REVIEW

**Comment by dhensby** [comment]
ping @camfindlay

File: docs/en/changelogs/index.md [hunk 1]
```diff
@@ -62,8 +62,4 @@ For information on how to upgrade to newer versions consult the [upgrading](/ins
  * [2.4.0-beta1](beta/2.4.0-beta1) - 29 January 2010
  * [2.4.0-alpha1](alpha/2.4.0-alpha1) - 11 November 2009
  * [2.3.11-rc1](rc/2.3.11-rc1) - 31 January 2011
- * [2.3.10-rc2](rc/2.3.10-rc2) - 20 December 2010
```
Comment (camfindlay) [inline docs/en/changelogs/index.md]: reason for removing these? the 2.3 history in the changelog should exist in the 2.4 docs too.

File: docs/en/howto/csv-import.md [hunk 2]
```diff
@@ -197,5 +197,4 @@ Sample implementation of a custom loader. Assumes a CSV-file in a certain format
 
 ## Related
 
-*  [api:CsvParser]
```
Comment (camfindlay) [inline docs/en/howto/csv-import.md]: reason for removal?

**Comment by camfindlay** [comment]
@SpiritLevel apart from a couple of questions looks good. Can you give me a comment on my points above.

File: docs/en/changelogs/index.md (hunk 1 above)
Comment (SpiritLevel) [inline docs/en/changelogs/index.md]: Howdy @camfindlay. The file rc/2.3.10-rc2.md doesn't exist. The other three were accidental deletions, sorry :) Now fixed.

File: docs/en/howto/csv-import.md (hunk 2 above)
Comment (SpiritLevel) [inline docs/en/howto/csv-import.md]: Fixed and tested.

**Comment by SpiritLevel** [comment]
@camfindlay Ok to merge now :) After all this work is done, I'll re-run the link checker to catch any that I miss.

File: docs/en/changelogs/index.md (hunk 1 above)
Comment (camfindlay) [inline docs/en/changelogs/index.md]: Is this release changelog in the 2.3 branch? Perhaps we can bubble it up if it's missing?

File: docs/en/changelogs/index.md [hunk 3]
```diff
@@ -62,8 +62,4 @@ For information on how to upgrade to newer versions consult the [upgrading](/ins
...
- * [2.3.10-rc1](rc/2.3.10-rc1) - 10 December 2010
```
Comment (camfindlay) [inline docs/en/changelogs/index.md]: is this meant to be removed?

File: docs/en/changelogs/index.md [hunk 4]
```diff
@@ -62,8 +62,4 @@ For information on how to upgrade to newer versions consult the [upgrading](/ins
...
- * [2.3.9-rc2](rc/2.3.9-rc2) - 4 November 2010
```
Comment (camfindlay) [inline docs/en/changelogs/index.md]: see above

File: docs/en/changelogs/index.md [hunk 5]
```diff
@@ -62,8 +62,4 @@ For information on how to upgrade to newer versions consult the [upgrading](/ins
...
- * [2.3.9-rc1](rc/2.3.9-rc1) - 1 November 2010
  * [2.3.8-rc1](rc/2.3.8-rc1) - 16 July 2010
```
Comment (camfindlay) [inline docs/en/changelogs/index.md]: see above

File: docs/en/changelogs/index.md (hunk 3 above)
Comment (SpiritLevel) [inline docs/en/changelogs/index.md]: Yes. rc/2.3.10-rc1.md is missing.

File: docs/en/changelogs/index.md (hunk 4 above)
Comment (SpiritLevel) [inline docs/en/changelogs/index.md]: Yes. rc/2.3.9-rc2.md is missing.

File: docs/en/changelogs/index.md (hunk 5 above)
Comment (SpiritLevel) [inline docs/en/changelogs/index.md]: Yes. rc/2.3.9-rc1.md is missing.

**Comment by camfindlay** [comment]
@SpiritLevel are those missing changelogs in any of the older branches at all?

File: docs/en/changelogs/index.md (hunk 5 above)
Comment (SpiritLevel) [inline docs/en/changelogs/index.md]: Can't see any docs or changelogs on 2.1 or 2.2 branches.

Comment (camfindlay) [inline docs/en/changelogs/index.md]: Must be from back when we moved from subversion to git and github! Will let this one through.

**Comment by camfindlay** [comment]
@dhensby this is ok to merge.

File: docs/en/changelogs/index.md (hunk 5 above)
Comment (SpiritLevel) [inline docs/en/changelogs/index.md]: I assume you have them in your SVN repo but probably not worth the trouble to find them...

**Comment by tractorcow** [comment]
Is this the right branch? I thought we were using post-2.4 for ongoing 2.4 maintenance.

**Comment by camfindlay** [comment]
We are indeed using the 2.4 branch see https://github.com/silverstripe/doc.silverstripe.org/blob/master/bin/update.sh#L49

**Comment by SpiritLevel** [comment]
Thanks for this as well @tractorcow ... and more coming for 2.4 :) 

If I come across non-docs things to fix in 2.4, should I submit a PR to post-2.4 or 2.4 ? 

What is the reason for the existence of post-2.4, as opposed to creating branches like 2.4.1, 2.4.2 etc....is it that you don't want to give the impression that semver was being follwed for 2.4 ?

**Comment by tractorcow** [comment]
Er, docs 2.4, non-docs don't bother? :P We aren't fixing 2.4.

**Comment by tractorcow** [comment]
post-2.4 is really "what would be 2.5 but we'll never release it"

**Pull request merged**